runtime/compiler/runtime/JitRuntime.cpp

```diff
@@ -108,18 +108,18 @@ J9::Recompilation::isAlreadyBeingCompiled(
    return TR::Recompilation::isAlreadyPreparedForRecompile(startPC);
    }
 
-void setDllSlip(char*CodeStart,char*CodeEnd,char*dllName, TR::Compilation * comp)
+void setDllSlip(const char *CodeStart, const char *CodeEnd, const char *dllName, TR::Compilation *comp)
```

Comment:
I notice there's [a call to `j9sl_open_shared_library` below](https://github.com/eclipse-openj9/openj9/blob/497c7358760bdb9a698284fe8409b1b4a4689295/runtime/compiler/runtime/JitRuntime.cpp#L124) that passes `dllName`as an argument.  It looks like [`j9sl_open_shared_library`](https://github.com/eclipse-openj9/openj9/blob/244201c2b45dcf1457117bd76c425e707a0bc79c/runtime/oti/j9port_generated.h#L571) is a macro defined in `j9port_generated.h`, which is defined in terms of [`sl_open_shared_library`](https://github.com/dylanjtuttle/omr/blob/83cb59850cf7c9da28183aeeaa7747908160835a/include_core/omrport.h#L2038), and it has a `char *` argument in that position.  I'm guessing that the compiler that's used for z/OS might very well complain about that, so you should probably change that to pass `const_cast<char *>dllName`

Reply:
Thanks for the catch! Fixed.

Reply:
Dylan @dylanjtuttle, it looks like I had typo. in my suggested change.  Parentheses are required around the operand of `const_cast`, so it should be `const_cast<char *>(dllName)`.  Without that builds on z/OS will fail.

Reply:
Sigh!  After your fix for that `const_cast`, the z/OS build reported another error [a few lines later in `JitRuntime.cpp`](https://github.com/eclipse-openj9/openj9/blob/776e08003e155c9da66c41ee609d564fec9d3bf0/runtime/compiler/runtime/JitRuntime.cpp#L138-L139):

```
15:40:03  "/u/jenkins/workspace/Build_JDK11_s390x_zos_Personal/openj9/runtime/compiler/runtime/JitRuntime.cpp", line 138.19: CCN5256 (S) A parameter of type "char *" cannot be initialized with an expression of type "const char *".
15:40:03  "/u/jenkins/workspace/Build_JDK11_s390x_zos_Personal/openj9/runtime/compiler/runtime/JitRuntime.cpp", line 138.19: CCN6205 (I) The error occurred while converting to parameter 1 of "void (char *, char *, void *, void *, char *, char *, char *)".
```

I think you'll need to change those two lines that pass `CodeStart` and `CodeEnd` to use `const_cast<char *>` as well.

Reply:
I probably should have spotted that before, my bad!